### PR TITLE
(PC-7482) profile & home: refetch /me on focus to update get credit update

### PR DESCRIPTION
**Pull request by AnoukHello**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-7482

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/home/pages/Home.tsx [hunk 1]
```diff
@@ -32,7 +32,12 @@ const statusBarHeight = getStatusBarHeight(true)
 export const Home: FunctionComponent = function () {
   const navigation = useNavigation<UseNavigationType>()
   const { params } = useRoute<UseRouteType<'Home'>>()
-  const { data: userInfos } = useUserProfileInfo()
+  const { data: userInfos, refetch } = useUserProfileInfo()
+  useFocusEffect(
```
Comment (kopax-polyconseil) [inline src/features/home/pages/Home.tsx:36]: `useFocusEffect(refetch)`

File: src/features/profile/pages/Profile.tsx [hunk 2]
```diff
@@ -45,7 +45,13 @@ const DEBOUNCE_TOGGLE_DELAY_MS = 5000
 export const Profile: React.FC = () => {
   const { dispatch: favoritesDispatch } = useFavoritesState()
   const { navigate } = useNavigation<UseNavigationType>()
-  const { data: user } = useUserProfileInfo()
+  const { data: user, refetch } = useUserProfileInfo()
+  useFocusEffect(
```
Comment (kopax-polyconseil) [inline src/features/profile/pages/Profile.tsx:49]: `useFocusEffect(refetch)`

File: src/features/home/pages/Home.tsx (hunk 1 above)
Comment (AnoukHello) [reply src/features/home/pages/Home.tsx:36]: C'est en faisant ça que j'ai l'erreur dont je te parlais sur slack : `It looks like you wrote 'useFocusEffect(async () => ...)' or returned a Promise. Instead, write the async function inside your effect and call it immediately...`

File: src/features/profile/pages/Profile.tsx (hunk 2 above)
Comment (AnoukHello) [reply src/features/profile/pages/Profile.tsx:49]: cf 1er commentaire

**Review by kopax-polyconseil — APPROVED** [review]

**Pull request merged**